server/plugin.go

```diff
@@ -335,15 +337,48 @@ func (p *Plugin) sendRefreshEvent(userID string) {
 	)
 }
 
-// HasProjectHook checks if the subscribed GitLab Project has a web hook
+// HasProjectHook checks if the subscribed GitLab Project or its parrent Group has a webhook
 // with a URL that matches the Mattermost Site URL.
 func (p *Plugin) HasProjectHook(user *gitlab.GitlabUserInfo, namespace string, project string) (bool, error) {
 	hooks, err := p.GitlabClient.GetProjectHooks(user, namespace, project)
 	if err != nil {
 		return false, errors.New("Unable to connect to GitLab")
 	}
 
+	//ignore error because many project won't be part of groups
+	hasGroupHook, _ := p.HasGroupHook(user, namespace)
+
+	if hasGroupHook {
+		return true, err
+	}
+
+	siteURL := *p.API.GetConfig().ServiceSettings.SiteURL
+	if siteURL == "" {
+		return false, emptySiteURLErr
+	}
```

Comment:
Instead of checking this everywhere, having the check in `OnActivate()` would allow admins to know right away that this plugin will have issue, if `SiteURL` is not set.